### PR TITLE
feat: add publish status to route in manager api

### PR DESCRIPTION
**Pull request by liuxiran**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues
issue #437 manage api
___
### Bugfix
- Description

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

1. add a Column to route, called `status`

1. `status = true`, means create a new route in both mysql and apisix;  `status=false`: means create a route only in mysql

1. modified the logic of editing and deleting route at the same time, 

1. add manager apis to support change route `status`

### CODE REVIEW

**Comment by moonming** [comment]
nice work @liuxiran 👍 
it will be better if you can add some test cases.

File: api/route/route.go
Comment (juzhiyuan) [inline api/route/route.go:40]: not sure if `r.GET("/apisix/admin/notexist/routes", isRouteExist)` will overrite this line, if it doesn't, then looks good to me :D

Comment (liuxiran) [reply api/route/route.go:40]: seems work well :)

**Review by juzhiyuan — APPROVED** [review]
Looks good to me from those logics, but still need a review from @nic-chen @gxthrj

**Comment by juzhiyuan** [comment]
@liuxiran CI failed, please take a look

**Comment by liuxiran** [comment]
> @liuxiran CI failed, please take a look

fixing :)

**Comment by liuxiran** [comment]
the reason for ci failed is the same with https://github.com/apache/apisix-dashboard/issues/452#issuecomment-688666125 , we updated the logic of routeGroup. this pr is depend on pr #460 

@juzhiyuan so sorry for the interdependence between these prs. I'll pay attention next time~!

**Comment by juzhiyuan** [comment]
@liuxiran ok, so just waiting for others' review :DD

**Comment by liuxiran** [comment]
> @liuxiran ok, so just waiting for others' review :DD

thanks for your time~

File: api/route/route.go
Comment (nic-chen) [inline api/route/route.go:65]: a little confused about `Create`, it should be `Update` or something here.

File: api/route/route_test.go
Comment (nic-chen) [inline api/route/route_test.go:43]: `offline` also requires test cases

File: api/script/db/schema.sql
Comment (nic-chen) [inline api/script/db/schema.sql:21]: deed a default value ​​for compatibility.

**Review by nic-chen — CHANGES_REQUESTED** [review]

**Comment by nic-chen** [comment]
api ci not pass

**Comment by juzhiyuan** [comment]
> the reason for ci failed is the same with [#452 (comment)](https://github.com/apache/apisix-dashboard/issues/452#issuecomment-688666125) , we updated the logic of routeGroup. this pr is depend on pr #460
> 
> @juzhiyuan so sorry for the interdependence between these prs. I'll pay attention next time~!

@nic-chen maybe this reason?

**Comment by nic-chen** [comment]
> > the reason for ci failed is the same with [#452 (comment)](https://github.com/apache/apisix-dashboard/issues/452#issuecomment-688666125) , we updated the logic of routeGroup. this pr is depend on pr #460
> 
> > 
> 
> > @juzhiyuan so sorry for the interdependence between these prs. I'll pay attention next time~!
> 
> 
> 
> @nic-chen maybe this reason?

maybe.so we should handle that pr first.

File: api/route/route.go
Comment (liuxiran) [reply api/route/route.go:65]: in this pr, we divided the process of creating a route into two steps：
create a route （without publish）only in mysql, and do `publish` to create a route in apisix.

so here we use `create`

File: api/script/db/schema.sql
Comment (liuxiran) [reply api/script/db/schema.sql]: done

File: api/route/route_test.go
Comment (liuxiran) [reply api/route/route_test.go:84]: done

**Comment by juzhiyuan** [comment]
@nic-chen https://github.com/apache/apisix-dashboard/pull/460 has been merged, please take a look.

File: api/route/route_test.go
Comment (nic-chen) [inline api/route/route_test.go:177]: why is true for offline ?

Comment (liuxiran) [reply api/route/route_test.go:177]: create a route online first, then we could  do `offline` :)

**Review by nic-chen — APPROVED** [review]

**Pull request merged**
